lxd/device/gpu_physical.go

```diff
@@ -102,6 +103,35 @@ func (d *gpuPhysical) Start() (*deviceConfig.RunConfig, error) {
 // Returns RunConfig populated with mount info required to pass the unix-char devices into the container.
 func (d *gpuPhysical) startContainer() (*deviceConfig.RunConfig, error) {
 	runConf := deviceConfig.RunConfig{}
+	if d.config["id"] != "" {
+		// Check if the id of the device match a CDI format.
+		cdiID, err := cdi.ToCDI(d.config["id"])
+		if err != nil {
+			return nil, err
+		}
+
+		// The cdiID can be empty if the id doesn't match the general CDI format, but this is not an error.
+		// and we allow the program to continue as the id could be a valid DRM card ID.
+		if !cdiID.Empty() {
+			if cdiID.Class == cdi.MIG {
+				return nil, fmt.Errorf("MIG GPU notation detected for a `physical` gputype device. Choose a `mig` gputype device instead.")
+			}
+
+			err = startWithCDISpec(d.state, d.inst, cdiID, &runConf)
```

Comment:
Why isn't startWithCDISpec not bound to the d var? Its part of the gpu device only right, and d is passed in.

Reply:
fixed